src/gwt/src/org/rstudio/studio/client/workbench/views/source/SourceWindowManager.java

```diff
@@ -944,9 +947,11 @@ private final native void saveWithPrompt(WindowEx satellite,
       satellite.rstudioSaveWithPrompt(item, onCompleted);
    }-*/;
 
-   private final native void saveAllUnsaved(WindowEx satellite, 
-         Command onCompleted) /*-{
-      satellite.rstudioSaveAllUnsaved(onCompleted);
+   private final native void saveUnsavedDocuments(WindowEx satellite,
+                                                  EditingTargetSavePredicate predicate,
+                                                  Command onCompleted)
+   /*-{
+      satellite.rstudioSaveUnsavedDocuments(predicate, onCompleted);
```

Comment:
It seems like the primary purpose of the predicate is to filter by ID, so if you wanted to mitigate any cross-window issues down the road you could pass POD like a JsArrayString of IDs instead. However I'll let you make the call on that based on your perception of the risk of the predicate evolving to become more complex.